vyper/ast/folding.py

```diff
@@ -172,20 +172,39 @@ def replace_user_defined_constants(vyper_module: vy_ast.Module) -> int:
             # annotation is not wrapped in `constant(...)`
             continue
 
-        changed_nodes += replace_constant(vyper_module, node.target.id, node.value, False)
+        # Extract typedef for propagation to type checker
+        node_annotation_arg = node.get("annotation.args")[0]
+        type = None
+        if node_annotation_arg.ast_type == "Subscript":
```

Comment:
I don't think you have to parse the type manually here, we have machinery for that